packages/auth/index.rn.ts

```diff
@@ -51,16 +51,37 @@ export { PhoneMultiFactorGenerator } from './src/platform_browser/mfa/assertions
 
 export { getReactNativePersistence };
 
+const NO_PERSISTENCE_WARNING = `
+You are initializing Firebase Auth for React Native without providing
+AsyncStorage. Auth state will default to memory persistence and will not
+persist between sessions. In order to persist auth state, install the package
+"@react-native-async-storage/async-storage" and provide it to
+initializeAuth:
+
```

Comment:
Not sure if the line break here is confusing but the string continues below with an example of the code the user should use. I used the template literal here so I could have more control over multiline formatting because I wanted to make sure the code example formatted nicely.